### DISCUSSION TITLE
Linking issues from PR sidebar (and vice versa) no longer works consistently

### DISCUSSION THREAD

**Discussion by MetRonnie** [community/community#44213] [Pull Requests]
### Select Topic Area

Bug

### Body

Ever since this new way of choosing linked PRs with the option to choose from different repos, searching for a PR number has not worked reliably

![image](https://user-images.githubusercontent.com/61982285/212308028-ba7a0ccf-1b97-4009-9feb-8160e371cc04.png)

Same for choosing a linked issue from a PR

![image](https://user-images.githubusercontent.com/61982285/212308076-09b32530-27b9-4754-8cb6-60a7ddc67ba5.png)

Trying the branch name or the PR/issue title does not work either. Only a limited number of PRs/issues show up if you clear the search box, and it seems that if the one you want isn't one of them, tough luck

**Comment by ristkof** [comment]
Upvoting this. I sometimes rename a branch for a pull request and I can not link it to the issue anymore. Annoying.
